internal/pipe/docker/docker.go

```diff
@@ -260,6 +269,70 @@ func dockerBuild(ctx *context.Context, root string, images, flags []string) erro
 	return nil
 }
 
+// in-memory store of docker registry to aviod relogin 
+// in case of multiple image tags
+var loggedInRegistry = make(map[string]bool)
+
+
+
+func dockerLogin(ctx *context.Context, dockerRegistry string) error {
+
+	log.WithField("docker registry", fmt.Sprintf("%s", dockerRegistry)).Info("log in to a registry")
+	if _, ok := loggedInRegistry[dockerRegistry]; ok {
+		log.Debug("already logged in!")
+		return nil
+	}
+	
+
+	var authToken string
+	userHome, err := homedir.Dir()
+	if err != nil {
+		return errors.Wrap(err, "failed to get home directroy")
+	}
+	dockerConfigPath := fmt.Sprintf("%s/config.json", userHome)
+
+	dockerConfigFile, err := os.Create(dockerConfigPath)
+	if err != nil {
+		return errors.Wrap(err, "failed to create docker registry config")
+	}
+	defer dockerConfigFile.Close()
+
+	switch dockerRegistry {
+	case "docker.pkg.github.com":
+		authToken = encodeToken("docker",  os.Getenv("GITHUB_TOKEN"))
```

Comment:
In theory the user may also decide to push images to a registry which isn't associated with the repository where the GitHub Action runs from. So while it's reasonable to expect `GITHUB_TOKEN` to be valid in such case, it will only have access to the origin registry - so some further filtering based on image path (not just the registry hostname) may be necessary here, to avoid logging into registries that the token doesn't have access to.

Reply:
@radeksimko Good Point. Prior to login, we could verify the image path using predefined variables `GITHUB_REPOSITORY` or `CI_REGISTRY_IMAGE`(GITLAB)